src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -76,8 +76,13 @@ func parseDuration(r *http.Request, key string) (time.Duration, error) {
 	}
 
 	// Try parsing as an integer value specifying seconds, the Prometheus default
-	if seconds, intErr := strconv.ParseInt(str, 10, 64); intErr == nil {
-		return time.Duration(seconds) * time.Second, nil
+	if seconds, intErr := strconv.ParseFloat(str, 64); intErr == nil {
+		ts := seconds * float64(time.Second)
+		if ts > float64(math.MaxInt64) || ts < float64(math.MinInt64) {
```

Comment:
Is it valid for duration to be negative in the prom context? If not, maybe check that it's positive instead of smaller than minInt64?

Reply:
Also, it might be worth storing the float version of maxInt64 as a const so you don't need to cast each time.

Reply:
This is what Prometheus checks, so tempted to just leave it. 

https://github.com/prometheus/prometheus/blob/master/web/api/v1/api.go#L1171

Reply:
> Also, it might be worth storing the float version of maxInt64 as a const so you don't need to cast each time.

Sounds good.